### ISSUE TITLE
stylelint message not showing interpolated values

### ISSUE THREAD

**Issue by anoopsinghbayes** [AndyOGo/stylelint-declaration-strict-value#142]
I am struggling with message 

```js
module.exports = {
  plugins: ["stylelint-declaration-strict-value"],
  rules: {
    "scale-unlimited/declaration-strict-value": [
      ["/color$/", "fill", "stroke"], {
        ignoreVariables:false,
        message: "Custom expected ${types} for \"${value}\" of \"${property}\"",
      }],
  },
};
```

and this what i see in vscode
![image](https://user-images.githubusercontent.com/6286517/109076379-1b5fe580-76c0-11eb-9a2b-9f9523f66437.png)

should'nt it show the interpolated values here 
or am i missing some thing

**Comment by AndyOGo** [comment]
Oh, that looks not as it should be.

I have a test here:
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/test/custom-message.js

may I ask you give more information about version of stylelint, this plugin, node?

**Comment by anoopsinghbayes** [comment]
"stylelint": "^13.11.0"
"stylelint-declaration-strict-value": "^1.7.7"
node :13.7.0

and did try this its the same issue
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/7f9475d9f701ac9a1657f33b1f59aa184ab9727a/test/custom-message.js#L15

**Comment by AndyOGo** [comment]
Thank.
I need to look into it.

**Comment by AndyOGo** [comment]
@anoopsinghbayes 
I added your case to my tests, but it succesfully runs 🤔 
https://github.com/AndyOGo/stylelint-declaration-strict-value/pull/143/files

Is there any other plugin you are using, or a VS code extension?

**Comment by anoopsinghbayes** [comment]
it does work without the message option
both in vscode and from the terminal
![image](https://user-images.githubusercontent.com/6286517/109084285-1c4b4400-76cd-11eb-9da9-569042c1db26.png)


![image](https://user-images.githubusercontent.com/6286517/109084311-2b31f680-76cd-11eb-9951-c2ddf0f22609.png)

**Comment by AndyOGo** [comment]
Great.

So I assume with the message option its broken in both, VScode extension and the terminal?

It would be very helpful if I could have a look myself and see what you are seeing.
Is your code public?

Btw. this code does the interpolation.
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/7f9475d9f701ac9a1657f33b1f59aa184ab9727a/src/lib/validation.ts#L218,L221

**Comment by anoopsinghbayes** [comment]
I don't have the repo public 
But I can put something together real quick tomorrow and will let you know

**Comment by AndyOGo** [comment]
That would be awesome, thank you

**Comment by AndyOGo** [comment]
I doublechecked the stylelint documentation.
Hope message is not a reserved secondary option.
https://stylelint.io/user-guide/configure#message

**Comment by AndyOGo** [comment]
@anoopsinghbayes Thanks for the issue.

I released version [`1.7.8`](https://github.com/AndyOGo/stylelint-declaration-strict-value/releases/tag/v1.7.8) which should fix this by undoing stylelint's internal custom message handling.

For details either have a look into the PR #143 or I also commented the official feature request [here](url)

**Comment by AndyOGo** [comment]
I created a follow up issue https://github.com/stylelint/stylelint/issues/5170

**Comment by anoopsinghbayes** [comment]
@AndyOGo your bug fix did the trick ,thanks a lot
i had a follow up question 
can i have separate message for different  selector 
for eg
/color$/ ==>message "use a color function xyz"
"padding==>message "use padding values abc"

i did look in the [scheme](https://github.com/AndyOGo/stylelint-declaration-strict-value#scheme) section ,didn't find anything that could help  am i missing something

**Comment by AndyOGo** [comment]
Thanks for your reply.
Glad to hear it works :)

Keep in mind though that it's a hacky workaround, wich messes with stylelint's internals.
This should become an official API as part of stylelint https://github.com/stylelint/stylelint/issues/5170

---

No, that's not possible as of now.
The only options which supports that is `ignoreValues`, because it needs it.

Why would you need that?

**Comment by anoopsinghbayes** [comment]
we have a design system with a bunch of utility classes 
like abc-padding-level1 abc-padding-level2 ,abc-margin-level1 etc for almost everything
we want to stop devs from using fixed value in the css either use the utility class or a scss function in place of fixed value 

for eg if some one does this

`.custom-padding-class{
  padding:6px; ///show stylelint error message "use the abc-padding-level1 class in the html or use padding- 
                          //function(level1)"
  color:red; //     show stylelint error message "use the abc-color-error class in the html or use color-function("error")
                     
}`


something like this

**Comment by AndyOGo** [comment]
Thanks, now I understand why you want to have that specific messages.

**Comment by AndyOGo** [comment]
Alright, I create a follow-up issue to elaborate the idea #146
Not that it will be implemented soon.
